### PR TITLE
Inject Template Name

### PR DESCRIPTION
**Pull request by mhuggins7278**
Inject the template name into the rendered template.

### CODE REVIEW

File: server.coffee [hunk 1]
```diff
@@ -308,7 +321,8 @@ validate_context_property = (property_name) ->
         log.error "missing expected property #{property_name} on context"
         ctx.error = new Error("missing expected property #{property_name} on context")
         # req is not serializable
-        log.debug "context: #{JSON.stringify _.omit(ctx, "req")}"
+        log.error "context: #{JSON.stringify _.omit(ctx, "req")}"
+        log.error ctx['rendered_template']
```
Comment (igroff) [inline server.coffee]: Why did you add this if it's already logging the contents of the context, and rendered_template is on the context?

File: server.coffee [hunk 2]
```diff
@@ -299,6 +299,19 @@ render_template_with_context = (ctx) ->
...
+inject_template_name = (ctx) ->
+  new Promise( (resolve, reject) ->
+      ###
+      if ctx.rendered_template.indexOf('EXEC sp_executesql') != -1
+        log.debug "******** Parameterized query detected do something smart here *************"
+      else
+      ###
+      arr = ctx.rendered_template.split('\n')
+      arr.unshift('-- ' + ctx.template_name)
+      ctx.rendered_template = arr.join('\n')
```
Comment (igroff) [inline server.coffee]: Why not just 
ctx.rendered_template = "-- #{ctx.template_name}\n" + ctx.rendered_template

File: server.coffee (hunk 1 above)
Comment (mhuggins7278) [reply server.coffee]: I changed it when debugging locally...i'll set it back didn't mean to commit that

File: server.coffee (hunk 2 above)
Comment (mhuggins7278) [reply server.coffee]: I was thinking moving forward that if we want to inject the template name in the sp_execute templates it would be easier to find the right place to do the injection this way..We can totally just go it as a string concat for now.

Comment (mhuggins7278) [reply server.coffee]: if we just want to do that couldn't we also just do it in the render_template_with_context method and not even bother with another function...until we need to?

Comment (igroff) [reply server.coffee]: 100%

Comment (mhuggins7278) [reply server.coffee]: ok I'll make the change and update the PR shortly.

**Comment by igroff** [comment]
Leaned up to just one line in render template.
